<launch>
    <!-- Arguments for detector node -->
    <arg name="name"/>
    <arg name="image_topic"/>
    <arg name="box_namespace" default="gt_box"/>
    <arg name="use_april_grid" default="true"/>
    <arg name="recording_bag_name" default=""/>

    <group ns="$(arg box_namespace)">
        <node name="$(arg name)" pkg="grand_tour_camera_detectors" type="grand_tour_camera_detector"
            output="screen">
            <param name="capability_group" value="Calibration"/>
            
            <param name="image_topic" value="$(arg image_topic)"/>
            <param name="show_extraction_video" value="false"/>
            <param name="recording_bag_name" value="$(arg recording_bag_name)"/>

            <!-- Parameters for apriltag -->
            <param name="grid_size_x" value="6" if="$(arg use_april_grid)"/>
            <param name="grid_size_y" value="6" if="$(arg use_april_grid)"/>
            <param name="tag_size" value="0.083" if="$(arg use_april_grid)"/>
            <param name="tag_spacing" value="0.3" if="$(arg use_april_grid)"/>

            <!-- Parameters for checkerboard -->
            <param name="rows" value="7" unless="$(arg use_april_grid)"/>
            <param name="cols" value="8" unless="$(arg use_april_grid)"/>
            <param name="row_spacing" value="0.08" unless="$(arg use_april_grid)"/>
            <param name="column_spacing" value="0.08" unless="$(arg use_april_grid)"/>
            <param name="use_april_grid" value="$(arg use_april_grid)"/>
        </node>
    </group> 
</launch>
